### ISSUE TITLE
Publishing a candidate does not transfer already-built docs

### ISSUE THREAD

**Issue by edsko** [haskell/hackage-server#70]
When publishing a candidate the resulting official package does not have any built docs, even if the original candidate did. The doc builder will attempt to rebuild them of course, but it might not be able to.

**Comment by expipiplus1** [comment]
As far as I can tell, candidates are published by getting the path to the tarball and "uploading" that again. So to fix this it would require finding the documentation tarball and "uploading" it here: https://github.com/haskell/hackage-server/blob/985c1ee842617abbdcf94b8e90b16960b8ab5f2e/Distribution/Server/Features/PackageCandidates.hs#L465

**Comment by gbaz** [comment]
@sitaochen did we look at this during gsoc?

**Comment by sitaochen** [comment]
Yea, the uploaded docs of candidates will also be there after they are published.

**Comment by expipiplus1** [comment]
oh great, so this can be closed?

**Comment by liskin** [comment]
I don't think this can be closed. I just published X11-1.10 which had docs uploaded to the candidate and the published release doesn't have the docs: https://hackage.haskell.org/package/X11-1.10
Can someone take a look what's going on under the hood?
